onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -144,7 +144,7 @@ class PlannerImpl {
   struct OrtValueInfo {
     const onnxruntime::NodeArg* p_def_site;  // the (unique) NodeArg corresponding to the MLValue
     int usecount = 0;                        // static reference-count
-    OrtValueIndex reused_buffer_index;       // index of original buffer to reuse
+    OrtValueIndex reused_buffer_index = -1;  // index of original buffer to reuse
   };
```

Comment:
maybe add a comment capturing why this needs to be set so it's not accidentally changed in the future.